flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBKeyedStateBackendBuilder.java

```diff
@@ -363,6 +363,9 @@ private static void checkAndCreateDirectory(File directory) throws IOException {
             IOUtils.closeQuietly(optionsContainer);
             ttlCompactFiltersManager.disposeAndClearRegisteredCompactionFactories();
             kvStateInformation.clear();
+            if (checkpointStrategy != null) {
+                checkpointStrategy.close();
```

Comment:
This call can potentially throw an exception preventing other cleanup actions. 
Those actions here are "safe" (e.g. use `closeQuietly` or `try/catch`). I think this call shouldn't throw any exception too. WDYT?

ditto: `RocksDBKeyedStateBackend`

Reply:
Couldn't agree more.  Has been fixed with `closeQuietly`.